mvp-spec/workflows.md

```diff
@@ -4,40 +4,77 @@ The workflows part of the Prebid Addressability Framework are:
 - User Id and Preferences
 - Selling ad slots
 
-## Id and User Preferences
+## User Id and Preferences
 
-### Generic workflow
+### Creating User Id and Preferences
 
-This workflow illustrates retrieving and creating/setting User Id and Preferences.
+This workflow illustrates a user first visit on a site part of the PAF network.
 
 ```mermaid
 sequenceDiagram
 participant U as User
 participant P as Participant's website
 participant O as Operator
 
-    U ->> P: visit site
+    U ->> P: Visit site
     
     P ->> O: Read User Id and Preferences
     O ->> P: No User Id and Preferences
     
+    P ->> P: Generate Id
```

Comment:
This might be misleading: it's **the operator** that generates IDs, not the publisher.
I recommend you add extra 2 calls (back and forth) to the operator to get the new ID.

```
P ->> O: Get new ID
O ->> O: Generate ID
O ->> P: New ID
```

The only thing is, readers should not get confused that at this stage, the id is **not** saved by the operator, even though it is generated.

Note: In practice, technically speaking, the ID is automatically generated by the operator and returned in the "no user id and preferences" response (line 22). But to be more comprehensive, an extra call might be clearer. I'll let you decide ;-)

Reply:
Thanks, updated.

> The only thing is, readers should not get confused that at this stage, the id is not saved by the operator, even though it is generated.

There's a note just below the diagram "Note that the User Id and Preferences are not persisted in case the user does not consent."

I think I should also update the diagram "Updating User Id and Preferences", taking into account the case where a new ID is being requested by the user. In that case, can you please confirm that the new ID is also generated by the operator?
```
P ->> O: Get new ID
O ->> O: Generate ID
O ->> P: New ID
```

Out of curiosity, why is not generated by the Publisher?

Reply:
Yes, you're right, the note is enough.

>. In that case, can you please confirm that the new ID is also generated by the operator?

yes

> Out of curiosity, why is not generated by the Publisher?

I have to say it was designed at a very early stage, but I think it is safer because:
- I think it makes sense to be a "centralized" ID generation, rather than to defer it to participants (even though I don't have a strong rationale to back it up :-) )
- we have more control over the update cycle of operators, than we have of publishers, in case we decide to change the algorithm to generate new IDs

Reply:
Agreed.

"Update Id and Preferences" workflow updated, and added consistency of verbs used across workflows.

Reply:
👌